### ISSUE TITLE
ReferenceError: require is not defined on production mode

### ISSUE THREAD

**Issue by Daggle24** [angular-architects/module-federation-plugin#301]
<img width="1512" alt="Captura de pantalla 2023-03-13 a las 0 28 11" src="https://user-images.githubusercontent.com/46680055/224582564-3499d8af-8dba-4d31-b583-de2a1f5ebd52.png">

When I upload my builded files to an s3 bucket, I get this error, I noticed the error comes from this line:

`import { loadRemoteEntry } from '@angular-architects/module-federation';

Promise.all([
    loadRemoteEntry({ type: 'module', remoteEntry: 'http://react-mf-bucket.s3-website-us-east-1.amazonaws.com/remoteEntry.js'})
])
.then(() => import('./bootstrap'))
.catch(err => console.error('error', err));`

any load method from '@angular-architects/module-federation' gives me this error on my production bucket.

But on my localhost environment it works without any problem. 

(Im guessing that is because im running my shell app with node instead of a static site.)

**Comment by Daggle24** [comment]
Sorry i misstyped the code. here is it:

```
import { loadRemoteEntry } from '@angular-architects/module-federation';

Promise.all([
loadRemoteEntry({ type: 'module', remoteEntry: 'http://react-mf-bucket.s3-website-us-east-1.amazonaws.com/remoteEntry.js'})
])
.then(() => import('./bootstrap'))
.catch(err => console.error('error', err));
```

**Comment by dzurrahman** [comment]
Same here, i got solution, but it's not good solution for production mode,
u can set `optimization: false` in angular.json config for production,

example : 

```
"production": {
  "optimization: false",
  "budgets": [
    {
      "type": "initial",
      "maximumWarning": "500kb",
      "maximumError": "1mb"
    },
    {
      "type": "anyComponentStyle",
      "maximumWarning": "2kb",
      "maximumError": "4kb"
    }
  ],
  "fileReplacements": [
    {
      "replace": "apps/greatdayhr-apps/src/environments/environment.ts",
      "with": "apps/greatdayhr-apps/src/environments/environment.prod.ts"
    }
  ],
  "outputHashing": "none",
  "customWebpackConfig": {
    "path": "apps/greatdayhr-apps/webpack.prod.config.js"
  }
},

```

**Comment by Daggle24** [comment]
Setting ```optimization:false``` fixed the issue.
however, i'll leave the issue open until we get a more targeted solution for this!

**Comment by dzurrahman** [comment]
agree, and there is information regarding the 'optimization' flag.

https://stackoverflow.com/questions/61630773/angular-difference-between-optimization-and-buildoptimizer-in-the-build-co

**Comment by dzurrahman** [comment]
update, try to change file .browserlist

```
# Firefox ESR
# not IE 11 # Angular supports IE 11 only as an opt-in. To opt-in, remove the 'not' prefix on this line.

Chrome >=79
ChromeAndroid >=79
Firefox >=70
Edge >=79
Safari >=14
iOS >=14
```

**Comment by ArioShaman** [comment]
> agree, and there is information regarding the 'optimization' flag.
> 
> https://stackoverflow.com/questions/61630773/angular-difference-between-optimization-and-buildoptimizer-in-the-build-co

Thx your solution really helps me.

In my case `"optimization: false"` doesn't work
